src/Microsoft.TestPlatform.Common/Filtering/FilterHelpers.cs

```diff
@@ -0,0 +1,115 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.VisualStudio.TestPlatform.Common.Filtering
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Globalization;
+    using System.Linq;
+    using System.Text;
+    using Microsoft.VisualStudio.TestPlatform.Common.Resources;
+
+    public static class FilterHelpers
+    {
+        private const char EscapePrefix = '%';  
+        private static readonly Dictionary<char, char> LookUpMap;  
+        private static readonly Dictionary<char, char> ReverseLookUpMap;    
+        private static readonly char[] SpecialCharacters;
+
+        static FilterHelpers()
+        {
+            LookUpMap = new Dictionary<char, char>()
+            {
+                { '%', '0'},
```

Comment:
Is this exhaustive list of special characters we expect ? If not, this won't scale as we can add just 2 more with current design as %9 will be the last supported value.

Reply:
I don't know if there's plan to expand operator/operation support for filter, but we can always use "%a", "%b", etc